src/js/background/background.js

```diff
@@ -25,6 +25,32 @@ browser.runtime.onInstalled.addListener(async (details) => {
   }
 });
 
+
+async function fetchRequestFromBackground(url, fetchMethod = "GET", body = null, opts=null) {
```

Comment:
Wouldn't it make sense to either move `fetchMethod` onto `opts`, or alternatively, to just accept the entire `init` parameter (i.e. the second parameter to `fetch`)?